src/torch/README.md

```diff
@@ -0,0 +1,178 @@
+# Notes for Design Discussion
+
+
+## File structure
+
+Currently the source is found here:
+```md
+SIRF
+├── src
+│   ├── torch
+│       ├── CMakeLists.txt
+│       ├── README.md
+│       └── SIRF_torch.py
+```
+Currently installed in:
+```md
+INSTALL
+├── python
+│   ├── sirf
+│       └── SIRF_torch.py
+```
+
+The classes are accessed as `sirf.SIRF_torch.TheClass`.
+
+## What is within `SIRF_torch`?
+
+Briefly going from the most abstracted:
+
+---
+
+So we have two functions for transferring from sirf to torch, and back. The reason for abstracting is because it is used often, as well as being somewhat prudent if this changes in the future. For example if we don't need to copy anymore.
+
+```python
+def sirf_to_torch(sirf_src, torch_dest, requires_grad=False):
+    if requires_grad:
+        return torch.tensor(sirf_src.as_array(), requires_grad=True).to(torch_dest.device)
+    else:
+        return torch.tensor(sirf_src.as_array()).to(torch_dest.device)
+```
+
+```python
+def torch_to_sirf(torch_src, sirf_dest): 
+    return sirf_dest.fill(torch_src.detach().cpu().numpy())
+```
+
+
+---
+Next we have subclassed `torch.autograd.Function`s these allow for PyTorch to make use of SIRF's acquisition models and objective functions.
+
+Generally they require a combinations of `torch_image`, `torch_measurements`, `sirf_image`, `sirf_measurements`, `sirf_acq_mdl`, and/or `sirf_obj_func`.
```

Comment:
As mentioned in https://github.com/SyneRBI/SIRF/pull/1305#pullrequestreview-2622728373, we need to generalise acq_model to other SIRF operations. Examples being `Reg.Resampler` and `STIR.AcquisitionSensitivityModel`, which have `forward` and `backward` operators.

Reply:
@KrisThielemans I changed to operators, and also abstracted the functionality to ensure the we are able to take care of different src and dest. We enforce [batch, channel, *src_shape] or [batch, *src_shape] as input, and [batch, channel, *dest_shape] or [batch, *dest_shape] as output.